front/lib/api/assistant/agent/agent_create.ts

```diff
@@ -0,0 +1,280 @@
+import { Op, Transaction } from "sequelize";
+
+import {
+  _buildAgentActionConfigurationTypeFromModel,
+  _buildAgentConfigurationTypeFromModel,
+  _buildAgentGenerationConfigurationTypeFromModel,
+} from "@app/lib/api/assistant/agent/agent_get";
+import { Authenticator } from "@app/lib/auth";
+import { front_sequelize } from "@app/lib/databases";
+import { DataSource, Workspace } from "@app/lib/models";
+import {
+  AgentDataSourceConfiguration,
+  AgentRetrievalConfiguration,
+} from "@app/lib/models/assistant/actions/retrieval";
+import {
+  AgentConfiguration,
+  AgentGenerationConfiguration,
+} from "@app/lib/models/assistant/agent";
+import { generateModelSId } from "@app/lib/utils";
+import {
+  AgentDataSourceConfigurationType,
+  isTemplatedQuery,
+  isTimeFrame,
+  RetrievalDataSourcesConfiguration,
+  RetrievalQuery,
+  RetrievalTimeframe,
+} from "@app/types/assistant/actions/retrieval";
+import {
+  AgentActionConfigurationType,
+  AgentConfigurationType,
+  AgentGenerationConfigurationType,
+} from "@app/types/assistant/agent";
+
+/**
+ * Create Agent Configuration
+ */
+export async function createAgentConfiguration(
+  auth: Authenticator,
+  {
+    name,
+    pictureUrl,
+  }: {
+    name: string;
+    pictureUrl: string;
+  }
+): Promise<AgentConfigurationType> {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Cannot create AgentConfiguration without workspace");
+  }
+
+  const agentConfig = await AgentConfiguration.create({
+    sId: generateModelSId(),
+    status: "active",
+    name: name,
+    pictureUrl: pictureUrl ?? null,
+    scope: "workspace",
+    workspaceId: owner.id,
+  });
+
+  return _buildAgentConfigurationTypeFromModel({
+    agent: agentConfig,
+  });
+}
+
+/**
+ * Create Agent Generation Configuration
+ */
+export async function createAgentGenerationConfiguration(
+  auth: Authenticator,
+  agentSid: string,
+  {
+    prompt,
+    modelProvider,
+    modelId,
+  }: {
+    prompt: string;
+    modelProvider: string;
+    modelId: string;
+  }
+): Promise<AgentGenerationConfigurationType> {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error(
+      "Cannot create AgentGenerationConfiguration: Workspace not found"
+    );
+  }
+
+  const agentConfig = await AgentConfiguration.findOne({
+    where: {
+      sId: agentSid,
+    },
+  });
+  if (!agentConfig) {
+    throw new Error(
+      "Cannot create AgentGenerationConfiguration: Agent not found"
+    );
+  }
+
+  const generation = await AgentGenerationConfiguration.create({
+    prompt: prompt,
+    modelProvider: modelProvider,
```

Comment:
to be consistent let's call these two fields 

`modelProviderId` and `modelModelId`